Mathlib/RingTheory/Ideal/Prod.lean

```diff
@@ -25,13 +25,26 @@ namespace Ideal
 def prod : Ideal (R × S) := I.comap (RingHom.fst R S) ⊓ J.comap (RingHom.snd R S)
 
 @[simp]
-theorem mem_prod {r : R} {s : S} : (⟨r, s⟩ : R × S) ∈ prod I J ↔ r ∈ I ∧ s ∈ J :=
+theorem coe_prod (I : Ideal R) (J : Ideal S) : (I.prod J : Set (R × S)) = (I : Set _) ×ˢ J :=
+  rfl
+
+@[simp]
+theorem mem_prod {x : R × S} : x ∈ prod I J ↔ x.1 ∈ I ∧ x.2 ∈ J :=
   Iff.rfl
 
 @[simp]
 theorem prod_top_top : prod (⊤ : Ideal R) (⊤ : Ideal S) = ⊤ :=
   Ideal.ext <| by simp
 
+@[simp]
+theorem prod_bot_bot : prod (⊥ : Ideal R) (⊥ : Ideal S) = ⊥ :=
+  SetLike.coe_injective <| Set.singleton_prod_singleton
+
+@[gcongr]
+theorem prod_mono {I₁ I₂ : Ideal R} {J₁ J₂ : Ideal S} (hI : I₁ ≤ I₂) (hJ : J₁ ≤ J₂) :
+    prod I₁ J₁ ≤ prod I₂ J₂ :=
```

Comment:
If I remember correctly, for gcongr to fire on `prod I J₁ ≤ prod I J₂`, we also need that special case explicitly stated (since it's a different pattern to match), so you might like to add that too

Reply:
At any rate, `Set` follows this pattern, so I've applied it.